backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt

```diff
@@ -207,4 +211,53 @@ class ShorkInterpreterControllerV1IT() : AbstractControllerTest() {
         assertEquals(HttpStatusCode.Created, result3.status)
         assertEquals("3", result3.bodyAsText())
     }
+
+    @Test
+    fun `test get all lobbies when there is only one and it's not initialized`() = runTest {
+        val result = client.get("/api/v1/lobbies")
+        val parsedLobbies = parseAllLobbies(result)
+
+        assertEquals(HttpStatusCode.OK, result.status)
+        assertEquals(1, parsedLobbies.size)
+        assertNotEquals(listOf("playerA", "playerB"), parsedLobbies[0].playersJoined)
+        assertEquals("NOT_STARTED", parsedLobbies[0].gameState)
+    }
+
+    @Test
+    fun `test get all lobbies as a list with players and games status for more than one lobby`() =
+        runTest {
+            client.post("/api/v1/lobby/0/code/playerA") {
+                contentType(ContentType.Application.Json)
+                setBody("someString")
+            }
+
+            client.post("/api/v1/lobby")
+            client.post("/api/v1/lobby")
+
+            client.post("/api/v1/lobby/2/code/playerA") {
+                contentType(ContentType.Application.Json)
+                setBody("someString")
+            }
+            client.post("/api/v1/lobby/2/code/playerB") {
+                contentType(ContentType.Application.Json)
+                setBody("someOtherString")
+            }
+
+            val result = client.get("/api/v1/lobbies")
+            val parsedLobbies = parseAllLobbies(result)
+
+            assertEquals(HttpStatusCode.OK, result.status)
+            assertEquals(3, parsedLobbies.size)
+
+            // Verify properties of each lobby
+            assertEquals("NOT_STARTED", parsedLobbies[0].gameState)
+            assertEquals(listOf("playerA"), parsedLobbies[0].playersJoined)
+
+            assertEquals(1L, parsedLobbies[1].lobbyId)
+            assertEquals("NOT_STARTED", parsedLobbies[1].gameState)
+            assertEquals(listOf("no player submission yet"), parsedLobbies[1].playersJoined)
+
+            assertEquals(listOf("playerA", "playerB"), parsedLobbies[2].playersJoined)
+            assertEquals("FINISHED", parsedLobbies[2].gameState)
+        }
```

Comment:
Please rewrite this to assume no ordering of the lobbies. The ordering is not defined.